NitroxModel/Discovery/GameInstallData.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using NitroxModel.Helper;
+
+namespace NitroxModel.Discovery
+{
+    /// <summary>
+    ///     Game definition data gathered from Steam's files.
+    /// </summary>
+    public class GameInstallData
+    {
+        private string managedDllsDir;
+        public string InstallDir { get; private set; }
+
+        public string ManagedDllsDir => managedDllsDir ??= Directory.EnumerateDirectories(InstallDir, "Managed", SearchOption.AllDirectories).FirstOrDefault();
+
+        private GameInstallData()
+        {
+        }
+
+        public GameInstallData(string installDir)
+        {
+            Validate.NotNull(installDir, $"Argument '{nameof(installDir)}' must not be null.");
+            InstallDir = installDir;
+        }
+
+        public static GameInstallData TryFrom(string path)
+        {
+            try
+            {
+                GameInstallData game = new();
+                XmlDocument xDoc = new();
+                xDoc.Load(path);
+                XmlNamespaceManager nsManager = new(xDoc.NameTable);
+                nsManager.AddNamespace("d", xDoc.DocumentElement.NamespaceURI);
+                foreach (XmlElement elem in xDoc.DocumentElement.SelectNodes("//d:PropertyGroup/*", nsManager))
+                {
+                    switch (elem.Name)
+                    {
+                        case "GameDir":
+                            game.InstallDir = elem.LastChild.Value;
+                            break;
+                        case "GameManagedDir":
+                            if (!Directory.Exists(elem.LastChild.Value))
+                            {
+                                throw new DirectoryNotFoundException();
+                            }
+                            game.managedDllsDir = elem.LastChild.Value;
+                            break;
+                    }
+                }
+                return game;
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        public void TrySave(string path)
+        {
+            static string PostfixBackslash(string text)
+            {
+                return text switch
+                {
+                    null => null,
+                    "" => "\\",
+                    _ => text.TrimEnd('\\') + '\\'
+                };
+            }
```

Comment:
Didn't have a good place to put it. Defining it on class as private would clutter class. So I just made it local method because I did need to call it multiple times.

Reply:
Yeah I understand your thinking, just wanted to make clear if this is something we can use in the future.